shotover-proxy/tests/kafka_int_tests/test_cases.rs

```diff
@@ -549,6 +551,159 @@ pub async fn produce_consume_partitions1_kafka_node_goes_down(
     }
 }
 
+pub async fn produce_consume_partitions1_shotover_nodes_go_down(
+    driver: KafkaDriver,
+    shotover_nodes_to_kill: &mut VecDeque<BinProcess>,
+    connection_builder: &KafkaConnectionBuilder,
+    topic_name: &str,
+) {
+    if driver.is_cpp() {
+        // Skip this test for CPP driver.
+        // While the cpp driver has some retry capabilities,
+        // in many cases it will mark a shotover node as down for a single failed request
+        // and then immediately return the error to the caller, without waiting the full timeout period,
+        // since it has no more nodes to attempt sending to.
+        //
+        // So we skip this test on the CPP driver to avoid flaky tests.
+        return;
+    }
+
+    {
+        let admin = connection_builder.connect_admin().await;
+        admin
+            .create_topics(&[NewTopic {
+                name: topic_name,
+                num_partitions: 1,
+                replication_factor: 3,
+            }])
+            .await;
+    }
+
+    {
+        let producer = connection_builder.connect_producer("all", 0).await;
+        // create an initial record to force kafka to create the topic if it doesnt yet exist
+        producer
+            .assert_produce(
+                Record {
+                    payload: "initial",
+                    topic_name,
+                    key: Some("Key".into()),
+                },
+                Some(0),
+            )
+            .await;
+
+        let mut consumer = connection_builder
+            .connect_consumer(
+                ConsumerConfig::consume_from_topic(topic_name.to_owned())
+                    .with_group("kafka_node_goes_down_test_group"),
+            )
+            .await;
+        consumer
+            .assert_consume(ExpectedResponse {
+                message: "initial".to_owned(),
+                key: Some("Key".to_owned()),
+                topic_name: topic_name.to_owned(),
+                offset: Some(0),
+            })
+            .await;
+
+        let num_nodes = shotover_nodes_to_kill.len();
+        // kill shotover node(s)
+        for _ in 0..num_nodes {
```

Comment:
can we just use a regular vec and do `for shotover_node in shotover_nodes_to_kill` ?